contrib/endpoints/src/api_manager/request_handler.cc

```diff
@@ -30,28 +30,34 @@ namespace google {
 namespace api_manager {
 
 void RequestHandler::Check(std::function<void(Status status)> continuation) {
-  if (context_) {
+  if (api_manager_->ConfigLoadingStatus().ok()) {
     InternalCheck(continuation);
   } else {
+    pending_check_callback_exist_ = true;
     api_manager_->AddPendingCheckReportCallback(
         [continuation, this](utils::Status status) {
           if (status.ok()) {
-            InitializeContext();
             InternalCheck(continuation);
           } else {
             continuation(status);
+            pending_check_callback_exist_ = false;
           }
         });
   }
 }
 
 void RequestHandler::InternalCheck(
     std::function<void(utils::Status status)> continuation) {
+  if (!context_) {
+    CreateRequestContext();
+  }
+
   auto interception = [continuation, this](utils::Status status) {
     if (status.ok() && context_->cloud_trace()) {
       context_->StartBackendSpanAndSetTraceContext();
     }
     continuation(status);
+    pending_check_callback_exist_ = false;
```

Comment:
no need for this.  this flag indicates if we are in pending callback for init_config_loading.  not for other kind of callbacks.  not need to worry other kind of callbacks.